### PR TITLE
Java caveat

### PR DESCRIPTION
**Pull request by vitorgalvao**
Refs https://github.com/caskroom/homebrew-cask/issues/16383.

If merged, we can start fixing:
- [ ] https://github.com/caskroom/homebrew-cask/issues/4885
- [ ] https://github.com/caskroom/homebrew-cask/search?utf8=%E2%9C%93&q=brew+cask+install+java

And others as we notice them.

### CODE REVIEW

**Comment by adityadalal924** [comment]
As a short term fix - this is good. We can use this until a real depends_on happens. Switching over should be trivial after the fact also, which is nice.

File: doc/CASK_LANGUAGE_REFERENCE.md [hunk 1]
```diff
@@ -219,6 +219,7 @@ The following methods may be called to generate standard warning messages:
 | --------------------------------- | ----------- |
 | `path_environment_variable(path)` | users should make sure `path` is in their `$PATH` environment variable
 | `zsh_path_helper(path)`           | zsh users must take additional steps to make sure `path` is in their `$PATH` environment variable
+| `depends_on_java(version)`        | users should make sure they have the specified version of java installed (defaults to `latest`)
```
Comment (jawshooah) [inline doc/CASK_LANGUAGE_REFERENCE.md]: Should probably specify that `version` must be an integer, not a full version spec (e.g. `6` or `7`)

Comment (vitorgalvao) [inline doc/CASK_LANGUAGE_REFERENCE.md]: How about like this?

Comment (jawshooah) [inline doc/CASK_LANGUAGE_REFERENCE.md]: :+1:

**Pull request merged**